larray/core/array.py

```diff
@@ -7925,6 +7925,59 @@ def apply(self, transform, *args, **kwargs):
                 # transpose back axis where it was
                 return res_arr.transpose(self.axes & res_arr.axes)
 
+    def apply_map(self, mapping, dtype=None):
```

Comment:
It would be nice to have an equivalent function for Axis and AxisCollection.

Reply:
Axis.replace already has that functionality (but is larger than this). For AxisCollection, I just created #782.

Reply:
see also #783

Reply:
given #783, do I leave the method as-is or do we defer adding apply_map until #783 is done. I would rather go for the former because I don't know when we will tackle #783 and I personally don't mind occasional deprecation messages, but our users might disagree 😄

Reply:
Leave the method as-is